src/upchunk.ts

```diff
@@ -217,19 +265,20 @@ export class UpChunk  {
         resolve();
       };
 
-      this.reader.readAsArrayBuffer(this.file.slice(start, start + length));
+      this.reader.readAsArrayBuffer(this.file.slice(this.nextChunkRangeStart, this.nextChunkRangeStart + length));
     });
   }
 
   private xhrPromise(options: XhrUrlConfig): Promise<XhrResponse> {
     const beforeSend = (xhrObject: XMLHttpRequest) => {
       xhrObject.upload.onprogress = (event: ProgressEvent) => {
-        const percentagePerChunk = 100 / this.totalChunks;
-        const sizePerChunk = percentagePerChunk * this.file.size;
-        const successfulPercentage = percentagePerChunk * this.chunkCount;
-        const currentChunkProgress = event.loaded / (event.total ?? sizePerChunk);
+        const remainingChunks = this.totalChunks - this.chunkCount;
+        // const remainingBytes = this.file.size-(this.nextChunkRangeStart+event.loaded);
```

Comment:
I can see how leaving this comment in could be useful. Intentional?

Reply:
Yes, that was intentional.  Not currently using the `remaningBytes` variable, but it was useful to have while debugging.